swupd/create_manifests.go

```diff
@@ -257,14 +257,51 @@ func processBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
 
 func addUnchangedManifests(appendTo *Manifest, appendFrom *Manifest) {
 	for _, f := range appendFrom.Files {
-		if f.findFileNameInSlice(appendTo.Files) == nil {
-			if f.Name == IndexBundle {
-				// this is generated new each time
-				continue
-			}
-			appendTo.Files = append(appendTo.Files, f)
+		if f.Name == IndexBundle {
+			// this is generated new each time
+			continue
+		}
+		if f.Type == TypeManifest &&
+			f.findFileNameInSlice(appendTo.Files) != nil {
+			continue
+		}
+		if f.Type == TypeIManifest &&
```

Comment:
Since you are doing the same operation whether it is a Manifest or an IManifest, why not just define a method on f that reports if it is any manifest type, `f.IsManifest()` or something?